src/process_cmd_line_arguments.hpp

```diff
@@ -0,0 +1,168 @@
+    std::endl <<
+    "Arguments:" << std::endl <<
+    "    YAML_CONFIG    path to a YAML configuration file." << std::endl <<
+    std::endl <<
+    "Options:" << std::endl <<
+    "    --from FROM_DOMAIN_ID    All data will be bridged from this domain ID.  " << std::endl <<
+    "                             This overrides any domain IDs set in the YAML file." <<
+    std::endl <<
+    "    --to TO_DOMAIN_ID        All data will be bridged to this domain ID.  " << std::endl <<
+    "                             This overrides any domain IDs set in the YAML file." <<
+    std::endl <<
+    "    --mode MODE_STR          Specify the bridge mode, valid values: '" << kCompressModeStr <<
+    "', '" << kDecompressModeStr << "' or '" << kNormalModeStr << "'.  " << std::endl <<
+    "                             This overrides any mode set in the YAML file." << std::endl <<
+    "                             Defaults to '" << kNormalModeStr << "'." <<
+    std::endl <<
+    "    --help, -h               Print this help message." << std::endl;
+}
+
+inline
+std::optional<size_t>
+parse_size_t_arg(const std::string & arg, const char * error_str)
+{
+  size_t value;
+  std::istringstream iss(arg);
+  iss >> value;
+  if (iss.fail() || !iss.eof()) {
+    std::cerr << "error: Failed to parse " << error_str << " '" <<
+      arg << "'" << std::endl;
+    return std::nullopt;
+  }
+  return value;
+}
+
+inline
+std::pair<std::optional<DomainBridgeConfig>, int>
+process_cmd_line_arguments(const std::vector<std::string> & args)
+{
+  if (args.size() < 2) {
+    std::cerr << "error: Expected YAML config file" << std::endl;
+    print_help();
+    return std::make_pair(std::nullopt, 1);
+  }
+  std::optional<size_t> from_domain_id;
+  std::optional<size_t> to_domain_id;
+  std::optional<domain_bridge::DomainBridgeOptions::Mode> mode;
+  std::optional<std::string> yaml_config;
+
+  for (auto it = ++args.cbegin() /*skip executable name*/; it != args.cend(); ++it) {
+    const auto & arg = *it;
+    if (arg == "-h" || arg == "--help") {
+      print_help();
+      return std::make_pair(std::nullopt, 0);
+    }
+    if (arg == "--from") {
+      if (from_domain_id) {
+        std::cerr << "error: --from option passed more than once" << std::endl;
+        print_help();
+        return std::make_pair(std::nullopt, 1);
+      }
+      ++it;
+      from_domain_id = parse_size_t_arg(*it, "FROM_DOMAIN_ID");
+      if (!from_domain_id) {
+        return std::make_pair(std::nullopt, 1);
+      }
+      continue;
+    }
+    if (arg == "--to") {
+      if (to_domain_id) {
+        std::cerr << "error: --to option passed more than once" << std::endl;
+        print_help();
+        return std::make_pair(std::nullopt, 1);
+      }
+      ++it;
+      to_domain_id = parse_size_t_arg(*it, "TO_DOMAIN_ID");
+      if (!to_domain_id) {
+        return std::make_pair(std::nullopt, 1);
+      }
+      continue;
+    }
+    if (arg == "--mode") {
```

Comment:
If we want to be extra careful, we could guard against the user entering the `mode` argument multiple times like we did for `from` and `to`

Reply:
Thanks! My idea was to also apply that logic here.
See https://github.com/ros2/domain_bridge/pull/35/commits/797c8b8d86f3e531dec5a366a5d6d528d4ef6ddb.